### DISCUSSION TITLE
FDRS with REYAX RYLR896 Lora Module

### DISCUSSION THREAD

**Discussion by javrolijk** [timmbogner/Farm-Data-Relay-System#96] [Ideas]
Dear all,

I have a few REYAX RYLR896 Lora Modules, which I'm trying to make work with the FDRS system and a BME280 as a sensor. Refer to the architecture below.
I don't think I can use the BME280 sensor example. Can anyone help with a hand with the programming?
Thanks in advance for any help.

![Screenshot 2022-08-15 215604](https://user-images.githubusercontent.com/70583691/184782303-e75325c1-ce31-413e-bf06-75b419c0cd34.png)

**Comment by timmbogner** [comment]
There seem to be a lot of UART LoRa radios on the market with varying methods of setup and usage. Expanding LoRa support to these varying devices will happen somewhat soon, hopefully. I have the E5 LoRa module from Seeed, which is on my list of things to add support for. Hopefully after that framework is laid, it will be pretty straightforward.
I did some research on the REYAX modules, and something interesting is that they have some addressing ability built in to their firmware.
I'm not sure whether this will be a blessing or a curse (or a non-issue). [This article](https://davywybiral.blogspot.com/2020/08/a-lesson-in-lora-module-p2p-standards.html) goes into some issues that might be faced in this realm.  I would need to see exactly what a packet from the REYAX module looks like to a "normal" LoRa module, (ie: one with no addressing). Do you have any other LoRa modules to experiment with?

**Reply by javrolijk** [reply]
Hi Tim.
Many thanks for the prompt reply. I have a project in mind which will consist of multiple light sensors (TSL2561) that will work as a Sky Quality Meter (SQM) in different locations to measure light pollution as it develops. All should send data to a gateway through Lora.

I have three types of Lora devices in my inventory for testing. These are as follows:

1. Reyax RYLR896 module
2. RFM95W LoraTM Transceiver
3. SX1276 Lora Transmitter

Distance between devices will be approximately 5-8km. Just wondering which Loras would work best with your FDRS system. I'm an electrical engineer but not a proficient programmer, so bear with me. 😊

Jairo

**Reply by timmbogner** [reply]
The more I read, the more it seems that these Reyax modules just aren't ideal for this purpose. They're made to communicate with each other using their own packet structure, and FDRS defines its own packet structure. I'm happy to help you continue to investigate, as I'm a bit curious. But unless there is an undocumented way to disable Reyax's addressing layer, I don't think they will be compatible with FDRS.

Luckily both the RFM95W and the SX1276 should work perfectly with FDRS.

**Reply by javrolijk** [reply]
Thanks for the info Tim, really appreciate it.
I think I'm leaning more towards using the SX1276 module. I was wondering if you could provide a hand with the light sensor (TSL2561) to make it work with the FDRS system. The following table provides how the luminance (lux) correlates to the SQM value.
![Screenshot_20220817-015600_OneNote](https://user-images.githubusercontent.com/70583691/185045333-adbe600e-f801-4489-9f8e-88d7b68402a1.jpg)

**Reply by timmbogner** [reply]
Yep, I'm looking into it. If you want a head start, you can combine the Adafruit TSL2561 library's example with the FDRS example.
